test/functional/page_objects/visual_builder_page.ts

```diff
@@ -750,4 +841,42 @@ export class VisualBuilderPageObject extends FtrService {
     const optionInput = await this.testSubjects.find(`filterRatio${optionType}Input`);
     await optionInput.type(query);
   }
+
+  public async toggleNewChartsLibraryWithDebug(enabled: boolean) {
+    await this.kibanaServer.uiSettings.update({
+      'visualization:visualize:legacyChartsLibrary': !enabled,
+    });
+    await this.elasticChart.setNewChartUiDebugFlag(enabled);
+  }
+
+  public async getChartDebugState(chartData?: DebugState) {
+    return chartData ?? (await this.elasticChart.getChartDebugData())!;
+  }
+
+  public async getXAxisTitle(chartData?: DebugState, nth: number = 0) {
+    const debugState = await this.getChartDebugState(chartData);
+    return debugState?.axes?.x[nth]?.title;
+  }
+
+  public async getLegendNames(chartData?: DebugState) {
+    const legendItems = (await this.getChartDebugState(chartData))?.legend?.items ?? [];
+    return legendItems.map(({ name }) => name);
+  }
+
+  public async getChartItems<ItemType extends ChartItem = 'areas'>(
```

Comment:
something goes wrong with that generic. You defined `ItemType` but in arguments you still use `ChartItem`. Not sure that we need generic here at all. Please remove